pandas/core/internals/managers.py

```diff
@@ -141,9 +141,37 @@ def __init__(
         if do_integrity_check:
             self._verify_integrity()
 
+        # Populate known_consolidate, blknos, and blklocs lazily
         self._known_consolidated = False
+        self._blknos = None
+        self._blklocs = None
 
-        self._rebuild_blknos_and_blklocs()
+    @property
+    def blknos(self):
+        """
+        Suppose we want to find the array corresponding to our i'th column.
+
+        blknos[i] identifies the block from self.blocks that contains this column.
+
+        blklocs[i] identifies the column of interest within
+        self.blocks[self.blknos[i]]
+        """
+        if self._blknos is None:
+            # Note: these can be altered by other BlockManager methods.
+            self._rebuild_blknos_and_blklocs()
```

Comment:
to make this more transparent, I would change the return signature of _rebuild_blknos_and_blklocs to return a tuple of (blknos, blklocs) and set them here. It is much more transparent.

Reply:
that turns out to get messy because of the way we update these in-place in some other methods.  so id like to skip this for now, and revisit it in the upcoming even-lazier pass

Reply:
ok.